apps/docs/content/guides/auth/third-party/auth0.mdx

```diff
@@ -0,0 +1,121 @@
+---
+id: 'auth-third-party-auth0'
+title: 'Auth0'
+subtitle: 'Use Auth0 with your Supabase project'
+---
+
+Auth0 can be used as a third-party authentication provider alongside Supabase Auth, or standalone, with your Supabase project.
+
+## Getting started
+
+1. First you need to add an integration to connect your Supabase project with your Auth0 tenant. You will need your tenant ID (and in some cases region ID).
+2. Add a new Third-party Auth integration in your project's [Authentication settings](/dashboard/project/_/settings/auth).
+3. Assign the `role: 'authenticated'` custom claim to all JWTs by using an Auth0 Action.
+4. Finally setup the Supabase client in your application.
+
+## Setup the Supabase client library
+
+<Tabs type="underlined" queryGroup="auth0-create-client">
+
+<TabPanel id="ts" label="TypeScript">
+
+```typescript
+import { createClient } from '@supabase/supabase-js'
+
```

Comment:
There should be a Python section too given it is official but not a blocker / high priority. Will open a thread in #team-contributors